android/app/src/main/java/com/couchbase/todo/db/DAO.java

```diff
@@ -282,6 +290,28 @@ public void deleteById(@NonNull String docId) {
         catch (CouchbaseLiteException e) { reportError(e); }
     }
 
+    public List getAllDocIds() throws CouchbaseLiteException {
+        List<String> idList = new ArrayList<>();
```

Comment:
Maybe move this down to line 297.  It's best not to have mutable things lying around before they are actually needed.